### PR TITLE
Refactor State & DataModel

### PR DESCRIPTION
**Pull request by Atralupus**
I changed the definition between state and datamodel more clearly

### CODE REVIEW

File: planet-clicker/Assets/_Script/Data/CountData.cs [hunk 1]
```diff
@@ -5,22 +5,21 @@ namespace _Script.Data
 {
     public class CountData : DataModel
     {
-        // NOTE: Explicitly set to lower case for compatibility.
-        public long count { get; private set; }
+        public long Count { get; private set; }
 
         public CountData(long c)
```
Comment (greymistcube) [inline planet-clicker/Assets/_Script/Data/CountData.cs:10]: As backward compatibility isn't much of a concern for this toy project, we can probably change this to `long count`.

File: planet-clicker/Assets/_Script/Action/AddCount.cs [hunk 2]
```diff
@@ -12,46 +12,39 @@ namespace _Script.Action
...
-
-        public AddCount()
-        {
-        }
+        private CountData _data;
 
         public AddCount(long count)
         {
-            _count = new CountData(count);
+            _data.Sum(count);
```
Comment (greymistcube) [inline planet-clicker/Assets/_Script/Action/AddCount.cs]: I think this should be `_data = new CountData(count)`?

File: planet-clicker/Assets/_Script/Data/CountData.cs [hunk 3]
```diff
@@ -5,22 +5,21 @@ namespace _Script.Data
...
         {
-            this.count = count;
         }
 
-        public CountData(Dictionary encoded)
-            : base(encoded)
+        public void Sum(long count)
         {
+            Count = Count + count;
         }
```
Comment (greymistcube) [inline planet-clicker/Assets/_Script/Data/CountData.cs]: We probably don't need this.

**Comment by greymistcube** [comment]
Is this not compatible with `development` branch? 😶

File: planet-clicker/Assets/_Script/Data/CountData.cs (hunk 1 above)
Comment (Atralupus) [reply planet-clicker/Assets/_Script/Data/CountData.cs:10]: yes good

**Review by greymistcube — APPROVED** [review]

File: planet-clicker/Assets/_Script/Action/AddCount.cs (hunk 2 above)
Comment (Atralupus) [reply planet-clicker/Assets/_Script/Action/AddCount.cs]: perfect point, and modified

**Comment by Atralupus** [comment]
> Is this not compatible with `development` branch? 😶

yes maybe not compatible

**Pull request merged**